### PR TITLE
[macOS] Add bundle launch code for older (pre 10.15) macOS versions.

### LINKED ISSUE #59781: MacOS : Godot 4.0 crashes on project opening or running
### Godot version

v4.0.alpha5.official.d7d528c15

### System information

Imac end 2012 i5 - MacOS 10.14.6- NVIDIA GeForce GTX 680MX

### Issue description

1) Create a new project in the project explorer and try to edit it : Godot will crash before opening the editor.
Or
2) Open a project by drag-and dropping the .project to the application : the editor will open. However, when trying to run the project, nothing will happen (no message except "--- Debugging process started ---")

The last error seen before the crash is :
```
ERROR: Could not create child process: /Applications/Godot.app
   at: create_process (drivers/unix/os_unix.cpp:390)
```

Might be related to issue #58985 as the VK_ERROR_INITIALIZATION_FAILED message appears, see below. However, the symptoms are a bit different, as I can't open the editor. As recommended in the ticket, I've tried deleting shader_cache without any effect.

When opening Godot, I get the following console message :

```
/Applications/Godot.app/Contents/MacOS/Godot
arguments
0: /Applications/Godot.app/Contents/MacOS/Godot
Current path: /Users/rataime
Godot Engine v4.0.alpha5.official.d7d528c15 - https://godotengine.org
Vulkan API 1.1.198 - Using Vulkan Device #0: NVIDIA - NVIDIA GeForce GTX 680MX
[mvk-error] VK_ERROR_INITIALIZATION_FAILED: Render pipeline compile failed (Error code 1):
Compiler encountered an internal error.
ERROR:  - Message Id Number: 0 | Message Id Name: 
	VK_ERROR_INITIALIZATION_FAILED: Render pipeline compile failed (Error code 1):
Compiler encountered an internal error.
	Objects - 1
		Object[0] - VK_OBJECT_TYPE_PIPELINE, Handle 140723555883008
   at: _debug_messenger_callback (drivers/vulkan/vulkan_context.cpp:159)
ERROR: vkCreateGraphicsPipelines failed with error -3 for shader 'ClusterRenderShaderRD:0'.
   at: render_pipeline_create (drivers/vulkan/rendering_device_vulkan.cpp:6596)
[mvk-error] VK_ERROR_INITIALIZATION_FAILED: Render pipeline compile failed (Error code 1):
Compiler encountered an internal error.
ERROR:  - Message Id Number: 0 | Message Id Name: 
	VK_ERROR_INITIALIZATION_FAILED: Render pipeline compile failed (Error code 1):
Compiler encountered an internal error.
	Objects - 1
		Object[0] - VK_OBJECT_TYPE_PIPELINE, Handle 140723563701760
   at: _debug_messenger_callback (drivers/vulkan/vulkan_context.cpp:159)
ERROR: vkCreateGraphicsPipelines failed with error -3 for shader 'ClusterRenderShaderRD:0'.
   at: render_pipeline_create (drivers/vulkan/rendering_device_vulkan.cpp:6596)
 
Registered camera Caméra FaceTime HD (intégrée) with id 1 position 0 at index 0
Editing project: /Users/rataime/Documents/Devel/Godot/Chem (::Users::rataime::Documents::Devel::Godot::Chem)
ERROR: Could not create child process: /Applications/Godot.app
   at: create_process (drivers/unix/os_unix.cpp:390)
```
Note that, until Registered camera, the messages don't prevent the project selector from running.

I've attached the result of the command `env MVK_DEBUG=1 MVK_CONFIG_LOG_LEVEL=4 /Applications/Godot.app/Contents/MacOS/Godot > mvk_log.txt 2>&1`
[mvk_log.txt](https://github.com/godotengine/godot/files/8396590/mvk_log.txt)

### Steps to reproduce

1) Create a new project in the project explorer and try to edit it
Or
2) Open a project by drag-and dropping the .project to the application, then run it

Reproduced on 4.0alpha5, 4.0alpha3, 4.0alpha1, but not on 3.4.4. 

### Minimal reproduction project

Create a new project, then try to edit it

### PR DESCRIPTION
**Pull request by bruvzg**
Make sure `create_instance` (which is calling `create_process` with bundle path, not an executable path) is handled correctly on older macOS versions.

Fixes #59781 (crash on project open, not a GPU issues, which is probably too old to work).

### CODE REVIEW

File: platform/osx/os_osx.mm [hunk 1]
```diff
@@ -313,21 +313,22 @@
...
-			NSMutableArray *arguments = [[NSMutableArray alloc] init];
-			for (const List<String>::Element *E = p_arguments.front(); E; E = E->next()) {
-				[arguments addObject:[NSString stringWithUTF8String:E->get().utf8().get_data()]];
-			}
+	// Use NSWorkspace if path is an .app bundle.
+	NSURL *url = [NSURL fileURLWithPath:@(p_path.utf8().get_data())];
+	NSBundle *bundle = [NSBundle bundleWithURL:url];
+	if (bundle) {
+		NSMutableArray *arguments = [[NSMutableArray alloc] init];
+		for (const List<String>::Element *E = p_arguments.front(); E; E = E->next()) {
```
Comment (akien-mga) [inline platform/osx/os_osx.mm:321]: Could use a range iterator I think.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Should this be cherrypicked for 3.x?

**Comment by bruvzg** [comment]
> Should this be cherrypicked for 3.x?

It won't case crash on 3.x, it does not have `create_instance`, and `exeute` will fallback to `fork`/`execvp`.

**Comment by glepache** [comment]
I didn't manage to run the artifacts of that PR (Vulkan library hell...), but will test in the next Alpha Release as I was the one having the linked issue.

**Pull request merged**